src/clients/entitlements/storage/in-memory/in-memory.cache.ts

```diff
@@ -0,0 +1,224 @@
+        if (userId) {
+          // that's user-targeted entitlement
+          const tenantUserEntitlements = this.ensureMapInMap(bundle.user_entitlements, tenantId);
+          const usersEntitlements = this.ensureArrayInMap(tenantUserEntitlements, userId);
+
+          usersEntitlements.push(this.parseExpirationTime(expirationDate));
+        } else {
+          // that's tenant-targeted entitlement
+          const tenantEntitlements = this.ensureArrayInMap(bundle.tenant_entitlements, tenantId);
+
+          tenantEntitlements.push(this.parseExpirationTime(expirationDate));
+        }
+      } else {
+        // TODO: issue warning here!
+      }
+    });
+
+    return bundlesMap;
+  }
+
+  private setupEntitlementsReadModel(src: BundlesSource): void {
+    const entitlementsReadModel: EntitlementsMap = new Map();
+
+    // iterating over bundles..
+    src.forEach((singleBundle) => {
+      // iterating over tenant&user entitlements
+      singleBundle.user_entitlements.forEach((usersOfTenantEntitlements, tenantId) => {
+        // iterating over per-user entitlements
+        usersOfTenantEntitlements.forEach((expTimes, userId) => {
+          const entitlementExpTime = pickExpTimestamp(expTimes);
+
+          singleBundle.features.forEach((feature) => {
+            entitlementsReadModel.set(getFeatureEntitlementKey(feature.key, tenantId, userId), entitlementExpTime);
+          });
+        });
+      });
+
+      // iterating over tenant entitlements
+      singleBundle.tenant_entitlements.forEach((expTimes, tenantId) => {
+        singleBundle.features.forEach((feature) => {
+          const entitlementExpTime = pickExpTimestamp(expTimes);
+
+          entitlementsReadModel.set(getFeatureEntitlementKey(feature.key, tenantId), entitlementExpTime);
+        });
+      });
+    });
+
+    this.nodeCache.set(ENTITLEMENTS_MAP_KEY, entitlementsReadModel);
+  }
+
+  private setupPermissionsReadModel(src: BundlesSource): void {
+    const permissionsReadModel: Map<string, Set<string>> = new Map();
+
+    src.forEach((singleBundle) => {
+      singleBundle.features.forEach((feature) => {
+        feature.permissions.forEach((permission) => {
+          this.ensureSetInMap(permissionsReadModel, permission).add(feature.key);
+        });
+      });
+    });
+
+    this.nodeCache.set(PERMISSIONS_MAP_KEY, permissionsReadModel);
+  }
+
+  private ensureSetInMap<K, T>(map: Map<K, Set<T>>, mapKey: K): Set<T> {
+    if (!map.has(mapKey)) {
+      map.set(mapKey, new Set());
+    }
+
+    return map.get(mapKey)!;
+  }
+
+  private ensureMapInMap<K, T extends Map<any, any>>(map: Map<K, T>, mapKey: K): T {
+    if (!map.has(mapKey)) {
+      map.set(mapKey, new Map() as T);
+    }
+
+    return map.get(mapKey)!;
+  }
+
+  private ensureArrayInMap<K, T>(map: Map<K, T[]>, mapKey: K): T[] {
+    if (!map.has(mapKey)) {
+      map.set(mapKey, []);
+    }
+
+    return map.get(mapKey)!;
+  }
+
+  private parseExpirationTime(time?: string): ExpirationTime {
```

Comment:
If an entitlement has no expiration time you'll get from backend `null`
so change your condition to 
`time !== null` or `time != undefined`